plugins/hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundDatabase.java

```diff
@@ -27,327 +27,289 @@
 
 public class GroundDatabase {
-  static final private Logger LOG = LoggerFactory.getLogger(GroundDatabase.class.getName());
+    static final private Logger LOG = LoggerFactory.getLogger(GroundDatabase.class.getName());
 
-  static final String DATABASE_NODE = "_DATABASE";
+    static final String DATABASE_NODE = "_DATABASE";
 
-  static final String DATABASE_TABLE_EDGE = "_DATABASE_TABLE";
+    static final String DATABASE_TABLE_EDGE = "_DATABASE_TABLE";
 
-  private static final List<String> EMPTY_PARENT_LIST = new ArrayList<String>();
+    private static final List<String> EMPTY_PARENT_LIST = new ArrayList<String>();
 
-  private GroundReadWrite groundReadWrite = null;
-  private GroundTable groundTable = null;
+    private GroundReadWrite groundReadWrite = null;
+    private GroundTable groundTable = null;
 
-  GroundDatabase(GroundReadWrite ground) {
-    groundReadWrite = ground;
-    groundTable = new GroundTable(ground);
-  }
+    GroundDatabase(GroundReadWrite ground) {
+        groundReadWrite = ground;
+        groundTable = new GroundTable(ground);
+    }
+
+    Node getNode(String dbName) throws GroundException {
+        LOG.debug("Fetching database node: " + dbName);
+        return groundReadWrite.getNode(dbName);
+    }
 
-  Node getNode(String dbName) throws GroundException {
-    try {
-      LOG.debug("Fetching database node: " + dbName);
-      return groundReadWrite.getNodeFactory().retrieveFromDatabase(dbName);
-    } catch (GroundException ge1) {
-      LOG.debug("Not found - Creating databsae node: {}", dbName);
+    Structure getNodeStructure(String dbName) throws GroundException {
+        return groundReadWrite.getStructure(dbName);
+    }
 
-      Node node = groundReadWrite.getNodeFactory().create(dbName);
-      Structure nodeStruct = groundReadWrite.getStructureFactory().create(node.getName());
-      LOG.debug("node structure created {}", nodeStruct);
-      return node;
+    Edge getEdge(NodeVersion nodeVersion) throws GroundException {
+        String edgeId = nodeVersion.getNodeId();
+        LOG.debug("Fetching database table edge: " + edgeId);
+        return groundReadWrite.getEdge(edgeId);
     }
-  }
-
-  Structure getNodeStructure(String dbName) throws GroundException {
-    try {
-      Node node = this.getNode(dbName);
-      return groundReadWrite.getStructureFactory().retrieveFromDatabase(dbName);
-    } catch (GroundException e) {
-      LOG.error("Unable to fetch database node structure");
-      throw e;
+
+    Structure getEdgeStructure(NodeVersion nodeVersion) throws GroundException {
+        try {
+            LOG.debug("Fetching database table edge structure: " + nodeVersion.getNodeId());
+            Edge edge = this.getEdge(nodeVersion);
+            return groundReadWrite.getStructure(edge.getName());
+        } catch (GroundException e) {
+            LOG.debug("Not found - database table edge structure: " + nodeVersion.getNodeId());
+            throw new GroundException(e);
```

Comment:
Again for RawStore implementation we make sure to use only MetastoreException. For internal methods we use GroundException